tests/jballerina-unit-test/src/test/resources/test-src/query/query-expr-with-anon-func-exprs.bal

```diff
@@ -0,0 +1,383 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+type Person record {|
+    string firstName;
+    string lastName;
+    int age;
+|};
+
+type Employee record {|
+    string firstName;
+    string lastName;
+    string dept;
+|};
+
+int y = 10;
+
+function testQueryWithAnonFuncExprInLetClause() {
+    int[] x = [1, 2];
+
+    (function () returns int)[] fn = from int m in x
+                                     let function () returns int func = function () returns int => m * y
+                                     select func;
+    assertEqual(fn.length(), 2);
+    function () returns int f1 = fn[0];
+    int value = f1();
+    assertEqual(value, 10);
+
+    f1 = fn[1];
+    value = f1();
+    assertEqual(value, 20);
+}
+
+function testQueryWithAnonFuncExprInWhereClause() {
+    int[] x = [1, 2];
+
+    int[] arr = from int m in x
+                let function () returns int func = function () returns int => m * y
```

Comment:
Try out a test with function that returns a function
eg:
`
function dummyFunc () returns (int) {
    return 1 + 2;
}

from...
let function () returns (function () returns (int)) func = function () returns (function () returns (int)) => dummyFunc
`